### PR TITLE
Feat/searches icon

### PR DESCRIPTION
**Pull request by anik-ghosh-au7**
Before submitting a pull request, please make sure the following is done:

- [x] Describe the proposed changes and how it'll improve the library experience.

The below-mentioned changes are applicable for the SearchBox components in react and vue

1. Default icons added for recent and popular searches
2. Support added to override default icons though - `recentSearchesIcon` or `popularSearchesIcon` props 
3. Support added to customize the icons through the `recent-search-icon` or 
 `popular-search-icon` keys of the innerClass prop

- [x] Please make sure that there are no linting errors in the code.

- [x] Add a demo video/gif/screenshot to explain how did you test the fix.
https://www.loom.com/share/c15a79932c7a4a7ab95b63395abbb808

- [ ] If it is a global change, try to add any side effects that it could have.

- [x] Create a PR to add/update the docs (if needed) at here.
https://github.com/appbaseio/Docs/pull/180

### CODE REVIEW

**Review by bietkul — CHANGES_REQUESTED** [review]
@anik-ghosh-au7 Why did you create new examples with just custom icons?
Can you please remove them?

File: packages/react-searchbox/src/components/SearchBox.js [hunk 1]
```diff
@@ -98,7 +99,7 @@ class SearchBox extends React.Component {
   get popularSuggestionsList() {
     const suggestions = this.componentInstance.suggestions;
     return (suggestions || []).filter(
-      suggestion => suggestion._popular_suggestion
+      suggestion => suggestion.source._popular_suggestion
```
Comment (bietkul) [inline packages/react-searchbox/src/components/SearchBox.js:103]: @anik-ghosh-au7 Why did you change it?

File: packages/react-searchbox/src/components/SearchBox.js [hunk 2]
```diff
@@ -112,7 +113,7 @@ class SearchBox extends React.Component {
     }
     const suggestions = this.componentInstance.suggestions;
     return (suggestions || []).filter(
-      suggestion => !suggestion._popular_suggestion
+      suggestion => !suggestion.source._popular_suggestion
```
Comment (bietkul) [inline packages/react-searchbox/src/components/SearchBox.js:117]: @anik-ghosh-au7 same here

File: packages/react-searchbox/src/components/SearchBox.js [hunk 3]
```diff
@@ -673,6 +706,8 @@ SearchBox.propTypes = {
   appbaseConfig: appbaseConfigDef,
   showDistinctSuggestions: bool,
   queryString: bool,
+  recentSearchesIcon: any,
```
Comment (bietkul) [inline packages/react-searchbox/src/components/SearchBox.js]: @anik-ghosh-au7 Use proper types

File: packages/react-searchbox/src/components/SearchBox.js [hunk 4]
```diff
@@ -534,17 +537,31 @@ class SearchBox extends React.Component {
...
                                 highlightedIndex,
                                 index
-                              )
+                              ),
+                              justifyContent: 'flex-start'
                             }}
                           >
+                            <div style={{ padding: '0 10px 0 0' }}>
+                              <CustomSvg
+                                iconId={`${sugg.id}-icon`}
```
Comment (bietkul) [inline packages/react-searchbox/src/components/SearchBox.js]: @anik-ghosh-au7 There is no `id` field present in the suggestion. How did you even test it?
Can you follow the code or log the suggestion object or just check the docs to find the suggestion object schema.

File: packages/react-searchbox/src/styles/CustomSvg.js [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+      style={{ fill: '#707070' }}
+      className={className}
+    >
+      <path d="M0 0h24v24H0z" fill="none" />
+      <path d="M16 6l2.29 2.29-4.88 4.88-4-4L2 16.59 3.41 18l6-6 4 4 6.3-6.29L22 12V6z" />
+    </svg>
+  );
+};
+
+export default CustomSvg;
```
Comment (bietkul) [inline packages/react-searchbox/src/styles/CustomSvg.js:51]: @anik-ghosh-au7 No prop type definition?

File: packages/react-searchbox/src/components/SearchBox.js [hunk 6]
```diff
@@ -563,25 +580,41 @@ class SearchBox extends React.Component {
...
-                              currentValue={currentValue}
-                              suggestion={sugg}
-                            />
-                          </li>
-                        ))}
+                      : (this.popularSuggestionsList || []).map(
+                          (sugg, index) => (
+                            <li
+                              {...getItemProps({ item: sugg })}
+                              key={`${index + this.suggestionsList + 1}-${
```
Comment (bietkul) [inline packages/react-searchbox/src/components/SearchBox.js]: @anik-ghosh-au7 What is `this.suggestionsList`? If it is an array then use `this.suggestionsList.length` for keys everywhere.

File: packages/vue-searchbox/src/components/SearchBox.jsx [hunk 7]
```diff
@@ -145,7 +146,7 @@ const SearchBox = {
 		getPopularSuggestionsList() {
 			const { suggestions } = this.getComponentInstance();
 			return (suggestions || []).filter(
-				suggestion => suggestion._popular_suggestion
+				suggestion => suggestion.source._popular_suggestion
```
Comment (bietkul) [inline packages/vue-searchbox/src/components/SearchBox.jsx:149]: @anik-ghosh-au7 Check react comments

File: packages/vue-searchbox/src/components/SearchBox.jsx [hunk 8]
```diff
@@ -158,7 +159,7 @@ const SearchBox = {
 			}
 			const { suggestions } = this.getComponentInstance();
 			return (suggestions || []).filter(
-				suggestion => !suggestion._popular_suggestion
+				suggestion => !suggestion.source._popular_suggestion
```
Comment (bietkul) [inline packages/vue-searchbox/src/components/SearchBox.jsx:162]: @anik-ghosh-au7 Check react comments

File: packages/vue-searchbox/src/components/SearchBox.jsx [hunk 9]
```diff
@@ -567,9 +583,23 @@ const SearchBox = {
...
 																	highlightedIndex,
 																	index + suggestionsList.length
-																)
+																),
+																justifyContent: 'flex-start'
 															}}
 														>
+															<div style={{padding: '0 10px 0 0'}}>
+																<CustomSvg
+																	iconId={`${sugg.id}-icon`}
```
Comment (bietkul) [inline packages/vue-searchbox/src/components/SearchBox.jsx]: @anik-ghosh-au7 Check react comments

File: packages/vue-searchbox/src/styles/CustomSvg.js [hunk 10]
```diff
@@ -0,0 +1,49 @@
+const CustomSvg = {
+	name: 'CustomSvg',
+	props: ['className', 'icon', 'type'],
```
Comment (bietkul) [inline packages/vue-searchbox/src/styles/CustomSvg.js]: @anik-ghosh-au7 Prop definition is missing

**Review by bietkul — CHANGES_REQUESTED** [review]
@anik-ghosh-au7 Share a video for `react` and `vue` with default icons.

File: packages/react-searchbox/src/components/SearchBox.js (hunk 1 above)
Comment (anik-ghosh-au7) [reply packages/react-searchbox/src/components/SearchBox.js:103]: @bietkul 

https://www.loom.com/share/55ec82016b6f47e48e4bfae8f6cda9c5

**Comment by anik-ghosh-au7** [comment]
@bietkul I have made the above changes

File: packages/react-searchbox/src/styles/CustomSvg.js [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+      <path d="M0 0h24v24H0z" fill="none" />
+      <path d="M16 6l2.29 2.29-4.88 4.88-4-4L2 16.59 3.41 18l6-6 4 4 6.3-6.29L22 12V6z" />
+    </svg>
+  );
+};
+
+CustomSvg.propTypes = {
+  iconId: string,
+  className: string,
+  icon: object,
```
Comment (bietkul) [inline packages/react-searchbox/src/styles/CustomSvg.js]: @anik-ghosh-au7 Use proper type

**Comment by anik-ghosh-au7** [comment]
demo with default icons (Searchbox-react) -
https://www.loom.com/share/45e93e1df1104ab8a2c2c3c09c6e27f3

**Review by bietkul — APPROVED** [review]
